### PR TITLE
update WDIO test to v5

### PR DESCRIPTION
**Pull request by saikrishna321**
## Proposed changes

Address https://github.com/appium/appium/issues/11912. Have updated sample-code and doc with WDIO vesion 5.X

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/en/about-appium/getting-started.md [hunk 1]
```diff
@@ -193,13 +193,12 @@ session:
...
 ```js
 // javascript
-client
-  .init()
-  .click("~App")
-  .click("~Alert Dialogs")
-  .back()
-  .back()
-  .end();
+    const elementId = await client.findElement("accessibility id","TextField1");
```
Comment (imurchie) [inline docs/en/about-appium/getting-started.md]: No need for indentation here.

File: docs/en/about-appium/getting-started.md [hunk 2]
```diff
@@ -231,13 +230,11 @@ const opts = {
...
-  .init()
-  .click("~App")
-  .click("~Alert Dialogs")
-  .back()
-  .back()
-  .end();
+const elementId = await client.findElement("accessibility id","TextField1");    client.elementSendKeys(elementId.ELEMENT, "Hello World!");
+const elementValue = await client.findElement("accessibility id","TextField1");
+await client.getElementAttribute(elementValue.ELEMENT,"value").then((attr) => {
+      assert.equal(attr,"Hello World!");
```
Comment (imurchie) [inline docs/en/about-appium/getting-started.md]: Two space indentation.

File: sample-code/javascript-webdriverio/package.json [hunk 3]
```diff
@@ -4,14 +4,14 @@
   "description": "Sample tests using webdriverio",
   "main": "index.js",
   "scripts": {
-    "test": "mocha --timeout 6000000 test/basic"
+    "test": "./node_modules/bin/mocha --timeout 6000000 test/basic"
```
Comment (imurchie) [inline sample-code/javascript-webdriverio/package.json]: This is not needed (or wanted). Commands run through the `"scripts"` section of the `package.json` have `./node_modules/.bin` added to the `PATH`. See https://docs.npmjs.com/cli/run-script

File: sample-code/javascript-webdriverio/test/basic/android-create-web-session.test.js [hunk 4]
```diff
@@ -1,29 +1,23 @@
...
-  before(function() {
-    client = webdriverio.remote(androidOptions);
-    return client.init();
+  before(async() => {
+    client = await webdriverio.remote(androidOptions);
   });
 
-  after(function() {
-    return client.end();
+  after(async() => {
```
Comment (imurchie) [inline sample-code/javascript-webdriverio/test/basic/android-create-web-session.test.js]: Do not use fat arrow functions within mocha apparatus (see https://mochajs.org/#arrow-functions).

File: sample-code/javascript-webdriverio/test/basic/android-create-web-session.test.js [hunk 5]
```diff
@@ -1,29 +1,23 @@
...
+    return await client.deleteSession();
   });
 
-  it("should create and destroy Android browser session", async function() {
+  it("should create and destroy Android browser session", async() => {
```
Comment (imurchie) [inline sample-code/javascript-webdriverio/test/basic/android-create-web-session.test.js]: I won't note the rest of these.

**Review by imurchie — APPROVED** [review]
Looks good to me! Thanks for doing this!

**Comment by rtmalone** [comment]
Minor thing...the text surrounding the code examples continues to use the phrase "Desired Capabilities". Does that continue to make sense after the key on the config has changed to just `capabilities`?

**Comment by dpgraham** [comment]
@rtmalone I think it's fine. `Desired Capabilities` is what they're called in the webdriver specification.

**Comment by adelphes** [comment]
Just a curiosity, but did anyone actually think to **read** the final page before merging this PR with the new code sample. As it still stands today, the text surrounding the code no longer matches anything with to do with the sample.

> In our case, we will simply tap into a few menus and then back out the way we came before ending the session:

```
// javascript

const field = await client.$("android.widget.EditText");
await field.setValue("Hello World!");
const value = await field.getText();
assert.equal(value, "Hello World!");
```

Absolutely nothing to do with menus here.

> What's going on here is that after creating a session and launching our app, we're instructing Appium to find an element in the app hierarchy and click on it.

Umm, that's not what's happening at all.

> Specifically, webdriverio has a convention where the ~ prefix means to find an element by its "accessbility id", which in the case of Android means an element's "content description". So we find and tap on these elements in order to navigate through the app's menu system. Then we can use the back() method to trigger the Android "back" behavior and get back to where we started before ending the session.

Again, nothing on this page relates to finding elements, the `~` prefix or pressing back.

Really, really curious how something so obvious gets missed in your reviews by multiple developers?

**Comment by imurchie** [comment]
How it is missed is probably because what is reviewed is the differences, not the full text.

**Pull request merged**